components/DatePicker/AvailabilityDatePicker.js

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect } from "react";
+import DatePicker from "react-datepicker";
+import { addMonths, endOfMonth } from "date-fns";
+
+import "react-datepicker/dist/react-datepicker.css";
+
+export function AvailabilityDatePicker({
+  onDateChange,
+  onShiftChange,
+  excludeDates,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const [selectedShift, setSelectedShift] = useState("");
+
+  const minDate = new Date();
+  const maxDate = endOfMonth(addMonths(new Date(), 1));
+
+  function handleClick(e) {
+    e.preventDefault();
+    setIsOpen(!isOpen);
+  }
+
+  useEffect(() => {
+    if (onDateChange) {
+      onDateChange(selectedDate);
+    }
+    if (onShiftChange) {
+      onShiftChange(selectedShift);
+    }
+  }, [selectedDate, selectedShift]);
```

Comment:
Setting state within `useEffect` hooks triggers unnecessary re-renders of all dependent components.

When you have the time to improve upon your code, [here's an excellent documentation](https://react.dev/learn/you-might-not-need-an-effect) of useEffect. It's about the fact that you might not need an effect at all. If you dig into it you'll find answers to your situation.